### PR TITLE
Define T_RUNTIME_FUNCTION in cross-bitness crossgen ARM32

### PR DESCRIPTION
**Pull request by echesakovMSFT**
In cross-bitness scenario `T_RUNTIME_FUNCTION` should be defined separately (`RUNTIME_FUNCTION` defined on amd64 host can not be used due to its different layout).
Also enable definitions of `RtlUnwindEx` `RtlVirtualUnwind` if host is either non-Windows or Windows/x86 and use host definitions on Windows/x64.

### CODE REVIEW

File: src/inc/crosscomp.h [hunk 1]
```diff
@@ -12,7 +12,7 @@
 #define CROSSBITNESS_COMPILE
 #endif
 
-#if defined(_X86_) && defined(_TARGET_ARM_)      // Host X86 managing ARM related code
+#if (defined(_X86_) || defined(_AMD64_)) && defined(_TARGET_ARM_) // Host X86 or AMD64 managing ARM related code
```
Comment (jkotas) [inline src/inc/crosscomp.h]: Can this be `!defined(_ARM_) && defined(_TARGET_ARM_)` ?

File: src/inc/crosscomp.h [hunk 2]
```diff
@@ -347,7 +356,7 @@ typedef struct _T_KNONVOLATILE_CONTEXT_POINTERS {
 
 } T_KNONVOLATILE_CONTEXT_POINTERS, *PT_KNONVOLATILE_CONTEXT_POINTERS;
 
-#else  // !(defined(_X86_) && defined(_TARGET_ARM_)) && !(defined(_AMD64_) && defined(_TARGET_ARM64_))
+#else  // !((defined(_X86_) || (defined(_AMD64_)) && defined(_TARGET_ARM_)) && !(defined(_AMD64_) && defined(_TARGET_ARM64_))
```
Comment (jkotas) [inline src/inc/crosscomp.h]: The comments on `#else` in these very long `if/elif/elif` sequences are not useful. I would just change this to `#else` without comment.

**Comment by mmitche** [comment]
@dotnet-bot test this please

**Comment by echesakovMSFT** [comment]
@jkotas Can you please take a look?

**Pull request merged**